x-pack/plugins/ml/public/application/components/saved_objects_warning/saved_objects_warning.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+
+import { EuiCallOut, EuiLink, EuiSpacer } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { JobType } from '../../../../common/types/saved_objects';
+import { useMlApiContext, useMlKibana } from '../../contexts/kibana';
+
+interface Props {
+  jobType?: JobType;
+}
+
+export const SavedObjectsWarning: FC<Props> = ({ jobType }) => {
+  const {
+    savedObjects: { initSavedObjects },
+  } = useMlApiContext();
+  const {
+    services: {
+      http: { basePath },
+    },
+  } = useMlKibana();
+
+  const [showWarning, setShowWarning] = useState(false);
+
+  useEffect(() => {
+    initSavedObjects(true)
+      .then(({ jobs }) => {
+        const missingJobs = jobType === undefined || jobs.some(({ type }) => type === jobType);
+        setShowWarning(missingJobs);
+      })
+      .catch(() => {
+        console.log('Saved object synchronization check could not be performed.'); // eslint-disable-line no-console
```

Comment:
is it worth showing a toast notification?

Reply:
no, this check will fail if the user does not have admin permission, so there should not be anything displayed on screen.
I added a log entry just so we can see that the check was attempted.